### PR TITLE
Include the option of McICA for the shortwave component of RRTMG

### PR DESCRIPTION
**Pull request by SallyDa**
I got stuck pretty much straight away and asked @skosukhin for help. He kindly agreed and decided to start from the original rrtmg files and to change those files as little as possible, to make it easier to update to a newer version of rrtmg in the future. Both the nomcica and mcica versions work and produce results very similar to those from @JoyMonteiro 's implementation of the nomcica rrtmg, except in the uppermost model level. I don't know why.

On the python side, I added a dictionary with the extra properties required for mcica, but I expect you might want to move this and add the * dimension. Let me know if you have any comments or suggestions and I can try to include them, or feel free to edit it yourselves!

Oh and I have no idea what the input parameter iplon in the mcica version of rrtmg is, but actually I don't think it does anything anyway...

### CODE REVIEW

**Comment by mcgibbon** [comment]
We need to get the code passing tests. This means a couple of things:

- go into the code directory and run `flake8`. If you don't have it installed, `pip install flake8` first. This will tell you any code style violations you need to fix.
- then, go into the tests directory and run `py.test`. Again, if you don't have it installed, `pip install pytest` first. This will run all our existing tests, make sure none break and if they do try to find out why. If tests break that are unrelated to RRTMG you may want to post here so we can help figure it out.
- Once you're sure no existing tests now fail, you can make new tests for this change. You should go into `test_components.py` and add a test class like

```
class TestRRTMGShortwaveMCICA(ComponentBaseColumn, ComponentBase3D):
    def get_component_instance(self):
        return RRTMGShortwave(mcica=True)
```

- Re-run the tests either with `py.test` or `python test_components.py`.  The first time you run these tests some will fail as they generate cached output to check against future times the tests are run. Run them a second time, and fix the code to pass any tests that fail this second time. This is only an absolutely bare test that makes sure the code runs without crashing, and that each time you run it it behaves the same.
- Also make an [example script](https://github.com/CliMT/climt/tree/develop/examples) that uses this option in a scientific context. Make sure to run it and see that it produces reasonable output (plots, etc.). This is a better test of whether the code is actually doing what it's intended to do, and not just running without errors.

File: climt/_components/rrtmg/sw/component.py [hunk 1]
```diff
@@ -396,54 +432,131 @@ def array_call(self, state):
         tendencies = initialize_numpy_arrays_with_properties(
             self.tendency_properties, state, self.input_properties
         )
+
+        num_reduced_g_intervals = self.num_reduced_g_intervals
+        mid_levels = state['air_pressure'].shape[0]
+        mcica_properties = {
```
Comment (mcgibbon) [inline climt/_components/rrtmg/sw/component.py]: Should these be set to zero, or taken as inputs and put into `input_properties`? Will this even work with a cloud water droplet radius of zero?

File: climt/_components/rrtmg/sw/component.py [hunk 2]
```diff
@@ -196,7 +200,10 @@ def __init__(
             ignore_day_of_year=False,
             facular_sunspot_amplitude=None,
             solar_variability_by_band=None,
-            aerosol_type='no_aerosol', **kwargs):
+            aerosol_type='no_aerosol',
```
Comment (mcgibbon) [inline climt/_components/rrtmg/sw/component.py]: You should add a description of what these additional keyword arguments do to the docstring of this method.

**Comment by JoyMonteiro** [comment]
Thanks for the PR! I will go through the code this week, but I wanted to mention that the reason there
is a change at the highest model level is that we have commented out lines in the code 
https://github.com/CliMT/climt/blob/707909e8f7019bcca77a48f6be99bac2bb447dd5/climt/_lib/rrtmg_sw/rrtmg_sw_rad.nomcica.f90#L813-L814

which cause RRTMG to violate energy conservation. This is done in the LW code as well.

I have no idea why they decided to zero out these values. This might be ok for GCMs (which is where RRTMG is meant to be used) which have numerical dissipation in the upper model levels, but it used to
destabilise the single column model.

**Comment by JoyMonteiro** [comment]
And before I forget, please also add your names to AUTHORS.rst!

**Comment by JoyMonteiro** [comment]
This looks great!! I will be happy to merge this once you update HISTORY and AUTHORS.

**Comment by JoyMonteiro** [comment]
Great! I'm going to ignore codedov. I will probably add some tests as well.
Thanks again @SallyDa @skosukhin, this is a very useful PR.

**Pull request merged**